napari/utils/validators.py

```diff
@@ -82,3 +84,34 @@ def func(obj):
                     )
 
     return func
+
+
+def pairwise(iterable: Iterable):
+    """Convert iterable to a zip object containing tuples of pairs along the
+    sequence.
+
+    Example:
```

Comment:
As a public function, this should follow the [numpydoc style for examples](https://numpydoc.readthedocs.io/en/latest/format.html#examples).

Or consider making this private (see https://github.com/napari/napari/pull/5036/files#r985971453).